### PR TITLE
Fix and enable RSA and X509Certificates tests on Unix

### LINKED ISSUE #1986: RSA/X509Certificates tests are segfaulting on Unix
Unfortunately it's hard to find what test/tests is/are failing due to the segfault.

If the tests would have the decency to fail on my machine the same way as they do in the CI build I'd be able to catch it in a debugger; but as it is the RSA tests are passing locally :smile:.

Since this failure is blocking opening up these libraries, I'm disabling the tests on Linux/OSX temporarily.

### PR DESCRIPTION
**Pull request by bartonjs**
Resolves #1986.

### CODE REVIEW

File: src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509CertificateReader.cs [hunk 1]
```diff
@@ -53,7 +58,7 @@ public AsymmetricAlgorithm PrivateKey
 
         public IntPtr Handle
         {
-            get { return IntPtr.Zero; }
+            get { return _cert == null ? IntPtr.Zero : _cert.DangerousGetHandle(); }
```
Comment (stephentoub) [inline src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509CertificateReader.cs]: Heh, this one made me smile :smile:

File: src/System.Security.Cryptography.X509Certificates/tests/TestData.cs [hunk 2]
```diff
@@ -370,5 +370,20 @@ internal static class TestData
             "29008DE8B851E2C30B6BF73F219BCE651E5972E62D651BA171D1DA9831A449D99" +
             "AF4E2F4B9EE3FD0991EF305ADDA633C44EB5E4979751280B3F54F9CCD561AC27D" +
             "3426BC6FF32E8E1AAF9F7C0150A726B").HexToByteArray();
+
+        internal static string NormalizeX500String(string expected)
+        {
+            if (StringComparer.Ordinal.Equals("Windows_NT", Environment.GetEnvironmentVariable("OS")))
```
Comment (stephentoub) [inline src/System.Security.Cryptography.X509Certificates/tests/TestData.cs]: Can you use Interop.PlatformDetection.cs instead? i.e. Interop.IsWindows?

**Comment by stephentoub** [comment]
One minor nit, otherwise LGTM.

File: src/System.Security.Cryptography.X509Certificates/tests/TestData.cs (hunk 2 above)
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/tests/TestData.cs]: Done.

**Pull request merged**